### ISSUE TITLE
In glpi 10, some input boxes will be displayed as source code in ticket content

### ISSUE THREAD

**Issue by triplility** [pluginsGLPI/formcreator#2521]
glpi 10 is great! and formcreator very import:)

After a while, I encountered a lot of problems, most of which I slowly solved by myself.

Ticket and FAQ under the form are not so convenient to connect with the native. 
Such as when I click "my requests for assistance" , everyone can see all form ticket in here, Although the person can't click to view details (because the corresponding native ticket doesn't have permission), but he just has to mouse over the title to see the details. I suspect it's something I set up wrong somewhere, but I really did not find it, I used a simple and rough way, the link to this page I changed to the native ticket view page(change the link from _/plugins/formcreator/front/issue.php_ to _/front/ticket.php_, everything is perfect)

But there are still very troublesome problems that I have not been able to solve:
I've noticed that some types of input boxes, when after created, in the contents of the native ticket, will be displayed as source code, such as the original bold and title will become as follow in ticket content.

``<h1>Form data</h1><h2>Information</h2><div><b>1) Categories : </b>Hardware / device / accessories</div><h2>further information</h2><div><b>2) accessories: : </b>
Wireless mouse</div>``

But not all types of input boxes, As long as I use these types in the question design of the form:
Textarea
Radios
Select
checkboxes
_and so on..._

**Comment by btry** [comment]
> Although the person can't click to view details (because the corresponding native ticket doesn't have permission)

I take note of this, I'll fix this as this is indeed a confidentiality issue.

I'll react soon about the remaining of your feedback.

**Comment by btry** [comment]
Hi

Try the PR #2524 : it should disable toltips for the items unreadable by the current user. I'm waiting for your feedback.

**Comment by btry** [comment]
> I've noticed that some types of input boxes, when after created, in the contents of the native ticket, will be displayed as source code, such as the original bold and title will become as follow in ticket content.

Maybe there is something specific to your instance as I you're alone to report such problem right now. The HTML sample you show is from a target using the ##FULLFORM## tag. 

Which type of target do you use ? target ticket or target change ?

**Comment by triplility** [comment]
> Hi
> 
> Try the PR #2524 : it should disable toltips for the items unreadable by the current user. I'm waiting for your feedback.

thank you for your update, regarding it, I try to copy and replace old file, it looks work on the list. 
**But.**..... the "seek assistance" page become blank. ( the helpdesk mode whether "service catalog extended" or "service catalog simplified" )

**Comment by triplility** [comment]
> > I've noticed that some types of input boxes, when after created, in the contents of the native ticket, will be displayed as source code, such as the original bold and title will become as follow in ticket content.
> 
> Maybe there is something specific to your instance as I you're alone to report such problem right now. The HTML sample you show is from a target using the ##FULLFORM## tag.
> 
> Which type of target do you use ? target ticket or target change ?

Yes, I use target ticket, I remember it should all be requests ticket type.

**Comment by btry** [comment]
You must not patch the plugin by copying the whole modified file.

You must 
- download the diff file, describing the change to do in the plugin. The diff is available by adding ".diff" at the end of the PR URL : https://github.com/pluginsGLPI/formcreator/pull/2524.diff
- locate in the root folder of Formcreator (ie: cd /var/lib/www/glpi/plugins/formcreator)
- simulate application of the patch with the patch binary tool : patch -p1 --dry-run < /path/to/diff-file.diff
- if no error occurs, then run again the patch tool, by removing the --dry-run argument.

**Comment by triplility** [comment]
Thanks I got it. but I'm not sure why, my file is not the same as yours, I checked my plugins version, it's 2.13.0-alpha.2 which looks like it's up to date. 

But in issue.class.php, your 770 line is my 792 line. And your 782 line as follow:

>                   $content = '';
>             }
>             $link = self::getFormURLWithID($id) . "&itemtype=".$data['raw']['itemtype'];
>             $link =  self::getFormURLWithID($data['id']);
>             $key = 'id';

but mine is like this and it's in line 803:

>                   $content = '';
>             }
>             $link = self::getFormURLWithID($data['id']);
> 
>             // Show "final" item id in the URL
>             if (Toolbox::isCommonDBTM($subItemtype)) {
>                $link .= "&" . $subItemtype::getForeignKeyField() . "=$id";
>             }
> 
>             $key = 'id';

So now, I have no idea if I need delete this plugin and reinstall it or is there anything else I can do to save it?

**Comment by btry** [comment]
Hi differences may appear if you read code from a different revision.  If you follow the instructions to apply the patch and focus on the issue it addresses then the difference between your copy and the repository don't matter.

**Comment by btry** [comment]
Hi

Did you succesfully installed the patch and validated it ?

**Comment by triplility** [comment]
> Hi
> 
> Did you succesfully installed the patch and validated it ?

Happy new year, I successfully installed the patch, but unfortunatally that it doesn't work. all the same before. I think maybe my plugins have something wrong, when I have time I will try to deleted and re-download / install this plugin.

**Comment by btry** [comment]
Can you identify why a ticket is not readable in your case ? (inaccessible entity for example)

**Comment by triplility** [comment]
> 

Do you mean that when I see another form, click on the unreadable prompt? it's : Item not found.

BTW, I really feel this change is enough:

In /plugin/formcreator/front/issue.php:

replace _Search::show('PluginFormcreatorIssue')_ to _Search::show('Ticket')_

**Comment by btry** [comment]
Hi

> Do you mean that when I see another form, click on the unreadable prompt? it's : Item not found.

Is your database populated from a fresh install of GLPI 10 or did you upgrade a previous database containing data ?

> BTW, I really feel this change is enough:

Your code modification is wrong. the page shall show assistance requests list, not tickets list.
